<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1030</width>
    <height>710</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1030</width>
    <height>700</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1030</width>
    <height>1000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>3DViewer_v2.0</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="projection_groupBox">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>230</y>
      <width>351</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>Тип проекции</string>
    </property>
    <widget class="QWidget" name="horizontalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>331</width>
       <height>21</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="projection_radioButtons">
      <item>
       <widget class="QRadioButton" name="parallel_radioButton">
        <property name="text">
         <string>Параллельная</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="central_radioButton">
        <property name="text">
         <string>Центральная</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="Widget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>69</y>
      <width>631</width>
      <height>631</height>
     </rect>
    </property>
   </widget>
   <widget class="QGroupBox" name="rotation_groupBox">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>10</y>
      <width>231</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Вращение</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>151</width>
       <height>103</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="rot_sliders">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <property name="leftMargin">
       <number>5</number>
      </property>
      <property name="topMargin">
       <number>3</number>
      </property>
      <property name="rightMargin">
       <number>5</number>
      </property>
      <item>
       <widget class="QSlider" name="rot_x_slider">
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>360</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="rot_y_slider">
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>360</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="rot_z_slider">
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>360</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>30</y>
       <width>41</width>
       <height>101</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="rot_lines">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLineEdit" name="rot_x_lineEdit">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="rot_y_lineEdit">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="rot_z_lineEdit">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>20</width>
       <height>101</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="rot_labels">
      <property name="leftMargin">
       <number>5</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="rot_x_label">
        <property name="text">
         <string>X</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="rot_y_label">
        <property name="text">
         <string>Y</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="rot_z_label">
        <property name="text">
         <string>Z</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="translating_groupBox">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>10</y>
      <width>111</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>Перемещение</string>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>20</width>
       <height>101</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="trans_labels">
      <property name="leftMargin">
       <number>5</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="trans_x_label">
        <property name="text">
         <string>X</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="trans_y_label">
        <property name="text">
         <string>Y</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="trans_z_label">
        <property name="text">
         <string>Z</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget_5">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>69</width>
       <height>101</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="trans_boxes">
      <item>
       <widget class="QDoubleSpinBox" name="trans_x_box">
        <property name="minimum">
         <double>-100.000000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="trans_y_box">
        <property name="minimum">
         <double>-100.000000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="trans_z_box">
        <property name="minimum">
         <double>-100.000000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="scaling_groupBox">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>160</y>
      <width>351</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>Масштабирование</string>
    </property>
    <widget class="QSlider" name="scale_slider">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>281</width>
       <height>24</height>
      </rect>
     </property>
     <property name="minimum">
      <number>5</number>
     </property>
     <property name="maximum">
      <number>20</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>10</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="scale_lineEdit">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>30</y>
       <width>39</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="edges_groupBox">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>300</y>
      <width>351</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string>Ребра</string>
    </property>
    <widget class="QWidget" name="horizontalLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>331</width>
       <height>21</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="edg_types_radioButtons">
      <item>
       <widget class="QRadioButton" name="solid_radioButton">
        <property name="text">
         <string>Сплошная</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="dashed_radioButton">
        <property name="text">
         <string>Пунктирная</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QSlider" name="edg_thickness_slider">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>60</y>
       <width>221</width>
       <height>24</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="edg_thickness_lineEdit">
     <property name="geometry">
      <rect>
       <x>298</x>
       <y>60</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>1</string>
     </property>
    </widget>
    <widget class="QLabel" name="edg_thickness_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Толщина</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_colorEdges">
     <property name="geometry">
      <rect>
       <x>9</x>
       <y>90</y>
       <width>281</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Выбор цвета</string>
     </property>
    </widget>
    <widget class="QFrame" name="frame_colorEdges">
     <property name="geometry">
      <rect>
       <x>298</x>
       <y>94</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="verticies_groupBox">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>440</y>
      <width>351</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string>Вершины</string>
    </property>
    <widget class="QWidget" name="horizontalLayoutWidget_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>331</width>
       <height>21</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="ver_types_radioButtons">
      <item>
       <widget class="QRadioButton" name="disable_radioButton">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Без вершин</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="square_radioButton">
        <property name="text">
         <string>Квадрат</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="circle_radioButton">
        <property name="text">
         <string>Круг</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QSlider" name="ver_size_slider">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>60</y>
       <width>221</width>
       <height>24</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="ver_size_lineEdit">
     <property name="geometry">
      <rect>
       <x>298</x>
       <y>60</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>5</string>
     </property>
    </widget>
    <widget class="QLabel" name="ver_size_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Размер</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_colorVerticies">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>281</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Выбор цвета</string>
     </property>
    </widget>
    <widget class="QFrame" name="frame_colorVerticies">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>95</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="vertices_label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Вершины</string>
    </property>
   </widget>
   <widget class="QLabel" name="filename_input_label">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>10</y>
      <width>581</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="edg_num_label">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>40</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="edges_label">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>40</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Грани</string>
    </property>
   </widget>
   <widget class="QLabel" name="ver_num_label">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>40</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="filename_label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Файл</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="background_groupBox">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>580</y>
      <width>351</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>Фон</string>
    </property>
    <widget class="QPushButton" name="pushButton_colorBackground">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>281</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Выбор цвета</string>
     </property>
    </widget>
    <widget class="QFrame" name="frame_colorBackground">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>35</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>660</y>
      <width>351</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="pushButton_makeBMP">
       <property name="text">
        <string>BMP</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_makeJPEG">
       <property name="text">
        <string>JPEG</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_reset">
       <property name="text">
        <string>Сброс модели</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pushButton_choiseFile">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>30</y>
      <width>113</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Выбор файла</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Widget</class>
   <extends>QOpenGLWidget</extends>
   <header>../engine/widget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
